Fix ServerSideRender in `/wp-admin/widgets.php`

kienstra opened the pull request:
<!--- Please summarize this PR in the title above -->

#### Changes
* Fixes a ServerSideRender error in `/wp-admin/widgets.php`:

![widget-blocke-editor](https://user-images.githubusercontent.com/4063887/133864635-5679201a-b3a2-4030-8f49-614b08ce0ae8.png)

```
{"code":"rest_invalid_param","message":"Invalid parameter(s): attributes","data":{"status":400,"params":{"attributes":"__internalWidgetId is not a valid property of Object."}
```

* It looks like `useBlockProps()` takes care of this.

Fixes [GF-3214](https://wpengine.atlassian.net/browse/GF-3214)
#### Testing instructions

kienstra committed:
Bump server-side-render to latest

kienstra committed:
Prevent the ServerSideRender error in /wp-admin/widgets.php

There was an error:
{"code":"rest_invalid_param","message":"Invalid parameter(s): attributes","data":{"status":400,"params":{"attributes":"__internalWidgetId is not a valid property of Object."}

kienstra committed:
Revert "Bump server-side-render to latest"

This reverts commit 0e75e1a23c527a912dffd8c79517d278d0b7f251.

kienstra committed:
Move useBlockProps() to Edit component

kienstra committed:
Mock  so there's ideally no Jest error

kienstra committed:
See if this fixes the failed Jest tests

Before, this didn't export
the store.

kienstra committed:
Empty commit to trigger CI

kienstra committed:
Empty commit to trigger CI

dreamwhisper reviewed (commented):
Unfortunately, I'm still experiencing the original error on the widgets screen with both old and new custom blocks. Still looking into it.

kienstra commented:
@dreamwhisper,
OK, thanks a lot for testing this.

kienstra committed:
Bump @wordpress/server-side-render to the latest

kienstra committed:
Merge branch 'develop' into fix/ssr-error-widgets-php

kienstra committed:
Downgrade @wordpress/server-side-render

There was an error in the GCB editor:
the block genesis-custom-blocks/foo
is not registered.

kienstra commented:
@dreamwhisper,
You're right, this PR doesn't fix it. I'm going to punt this until a later release.
